hudi-common/src/test/java/org/apache/hudi/common/testutils/HoodieCommonTestHarness.java

```diff
@@ -52,6 +53,24 @@ protected void initPath() {
     }
   }
 
+  /**
+   * Initializes a test data generator which used to generate test datas.
+   *
+   */
+  protected void initTestDataGenerator() {
```

Comment:
Why do we need to move this code snippet from `HoodieClientTestHarness` in this PR?

Reply:
This can be reuse by both Spark and Flink client test harness and only depend on hudi's code.